express-notes-api/index.js

```diff
@@ -0,0 +1,139 @@
+  console.error(newNote.content);
+  if (req.body.content === undefined) {
+    const error = function newError() {
+      return { error: 'content is a required field' };
+    };
+    console.error(error());
+    res.status(400).json(error());
+  } else {
+    const id = dataJSON.nextId;
+    newNote.id = id;
+    dataJSON.notes[id] = newNote;
+    dataJSON.nextId++;
+    console.error(dataJSON);
+    const data = new Uint8Array(Buffer.from(JSON.stringify(dataJSON, null, 2)));
+    fs.writeFile('data.json', data, 'utf-8', err => {
+      if (err) {
+        const error = function newError() {
+          return { error: 'An unexpected error occurred.' };
+        };
+        console.error(error());
+        res.status(500).json(error());
+      } else {
+        res.status(201).json(newNote);
+      }
+    });
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const currentId = req.params.id;
+  if (currentId < 0) {
+    const error = function newError() {
+      return { error: 'id must be a positive integer' };
+    };
+    console.error(error);
+    res.status(400).json(error);
+  } else if (dataJSON.notes[currentId] === undefined) {
+    const error = function newError() {
+      return { error: `cannot find note with id ${currentId}` };
+    };
+    console.error(error());
+    res.status(404).json(error());
+  } else {
+    delete dataJSON.notes[currentId];
+    const data = new Uint8Array(Buffer.from(JSON.stringify(dataJSON, null, 2)));
+    fs.writeFile('data.json', data, 'utf-8', err => {
+      if (err) {
+        const error = function newError() {
+          return { error: 'An unexpected error occurred.' };
+        };
+        console.error(error());
+        res.status(500).json(error());
+      } else {
+        res.status(204).send();
+      }
+    });
+  }
+});
+
+app.put('/api/notes/:id', (req, res) => {
+  const currentId = req.params.id;
+  if (currentId < 0) {
+    const error = function newError() {
+      return { error: 'id must be a positive integer' };
+    };
+    console.error(error());
+    res.status(400).json(error());
+  } else if (req.body.content === undefined) {
+    const error = function newError() {
+      return { error: 'content is a required field' };
+    };
+    console.error(error());
+    res.status(400).json(error());
+  } else if (dataJSON.notes[currentId] === undefined) {
+    const error = function newError() {
+      return { error: `cannot find note with id ${currentId}` };
+    };
+    console.error(error());
+    res.status(400).json(error());
+  } else {
+    const updatedNote = req.body;
+    updatedNote.id = currentId;
+    dataJSON.notes[currentId] = updatedNote;
+    const data = new Uint8Array(Buffer.from(JSON.stringify(dataJSON, null, 2)));
+    fs.writeFile('data.json', data, 'utf-8', err => {
+      if (err) {
+        const error = function newError() {
+          return { error: 'An unexpected error occurred.' };
+        };
+        console.error(error());
```

Comment:
Whenever your code runs into an `err` like this, it should `console.error(err)` so that you can see the stack trace of the problem for debugging purposes. The generic error message is meant for the client, but as the back end developer you want all of the gory details.